dmoj/executors/RUST.py

```diff
@@ -86,6 +87,12 @@ def get_shared_target(self):
                 # We intentionally don't clean this directory up at any point, since we can re-use it.
                 return self.shared_target
 
+    def get_fs(self) -> List[FilesystemAccessRule]:
+        assert self._executable is not None
+        # Under landlock we need this for execve to work.
+        # We use `self._executable` because it is copied when caching executors, but other properties are not.
+        return super().get_fs() + [ExactFile(self._executable)]
```

Comment:
The fact that both `get_executable` and `get_compiled_file` exist but return possibly different values is an abomination, and we should fix this. You don't have to do so in this PR, but please leave a comment to the effect of this being a hack to work around that.

Reply:
The problem isn't because `get_executable` and `get_compiled_file` both exist. Indeed, I expect them to return the same file for RUST.

The reason they don't is because the cached executor isn't a full copy of the original. Instead, it simply sets `_executable` and `_dir` on the returned object, and nothing else.

For rust, that means that we try and lock a new directory, which is obviously wrong.

To bypass it I use `_executable` here, but the proper fix may be instead to look into the caching.

Here's the relevant line: https://github.com/DMOJ/judge-server/blob/master/dmoj/executors/compiled_executor.py#L58

Also, as for `get_executable` vs `get_compiled_file`, check out Java's usage of the methods:

https://github.com/DMOJ/judge-server/blob/master/dmoj/executors/java_executor.py#L82

Maybe these exist because Java's compiled file isn't the executable?